cmd/controller-manager/app/serve.go

```diff
@@ -32,11 +31,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/configz"
 )
 
-type serveFunc func(handler http.Handler, shutdownTimeout time.Duration, stopCh <-chan struct{}) error
-
-// Serve creates a base handler chain for a controller manager. It runs the
-// the chain with the given serveFunc.
-func Serve(c *CompletedConfig, serveFunc serveFunc, stopCh <-chan struct{}) error {
+// DefaultBuildHandlerChain takes in CompletedConfig and returns a handler.
+func DefaultBuildHandlerChain(c *CompletedConfig) http.Handler {
```

Comment:
We're trying to match this: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/config.go#L130 so that we'll have choices about using this instead.

Reply:
Thanks for your explanation. And will think about tomorrow morning。

Reply:
@deads2k PTAL